lib/datadog/statsd/single_thread_sender.rb

```diff
@@ -7,10 +7,12 @@ class Statsd
     # It is using current Process.PID to check it is the result of a recent fork
     # and it is reseting the MessageBuffer if that's the case.
     class SingleThreadSender
-      def initialize(message_buffer, logger: nil, flush_interval: nil)
+      def initialize(message_buffer, logger: nil, flush_interval: nil, queue_size: 1)
         @message_buffer = message_buffer
         @logger = logger
         @mx = Mutex.new
+        @queue_size = queue_size
+        @queue = []
```

Comment:
Do you think we should consider naming these `message_queue`, `message_queue_size` and `drain_message_queue` instead, to mimic the other sender implementation?

Reply:
It's not a strong opinion, it might even be more confusing (it is yet another kind of queue, different from what exists in the other implementation).

In all cases, a one-line comment on the declaration
```
queue = []
```
might be helpful for a future reader.

Reply:
I think changing the names to match is a good idea. Even though the queue implementations is different, the idea was to make the single and the multi threaded senders more compatible so that we have less things to reason about.

Reply:
> In all cases, a one-line comment on the declaration
>
> ```
> queue = []
> ```
>
> might be helpful for a future reader.

I'm not sure I understand. Could you please tell me more about what you'd expect on such a comment?

Reply:
Yeah I agree about the difference of queue implementation/usage, it's a reason I hesitated to comment this... Less reasoning is probably better though. Please go with `@message_queue_size` if you don't mind.

> Could you please tell me more about what you would find helpful to see in such a comment?

Now named `@message_queue`, I don't feel it needs a comment anymore, please ignore my request :)

Reply:
Hm, indeed. It's fine as it is now, thanks for the changes already. 🙇 
I'll track the discrepancy in a separate task and I might send a small refactor change later on.